### ISSUE TITLE
Update AuditJS parser

### ISSUE THREAD

**Issue by ThiefaineM** [DefectDojo/django-DefectDojo#6154]
The AuditJS parser only supports purl that looks like : pkg:type/PACKAGE_NAME@PACKAGE_VERSION
It create errors when the purl has a namespace.

**Solution**
A solution could be to support all types of purl : pkg:type/namespace/name@version?qualifiers
If only component_name and component-version are considered worthy to save, by counting the number of "/" and splitting we can get the information.
Else the new component name could be namespace:name as in Dependency-Check parser.

**Comment by damiencarol** [comment]
@ThiefaineM we rely on `https://pypi.org/project/purl/` project today to parse PURL. Do you have an ex of errors when loading an AuditJS report?

**Comment by ThiefaineM** [comment]
The issue here does not come from ``` https://pypi.org/project/purl/ ``` but this part of the code:
```
for dependency in data:
            # reading package name in format pkg:npm/PACKAGE_NAME@PACKAGE_VERSION
            if "coordinates" in dependency:
                file_path = dependency["coordinates"]
                component_name, component_version = file_path.split('/')[1].split('@')
```
When there is a namespace the purl contains two '/'. So the field file_path.split('/')[1] only contains the namespace and thus no '@'. The second split will return only one value. It creates an error because it is expecting two value.

**Comment by damiencarol** [comment]
@ThiefaineM do you have a report that generate these errors to test? (you can obfuscate it if needed)

**Comment by damiencarol** [comment]
Fixed by #6596
